crates/future_metrics/src/lib.rs

```diff
@@ -0,0 +1,189 @@
+use {
+    metrics::{Counter, Histogram, Key, Label, Level, Metadata},
+    std::{
+        future::Future,
+        pin::Pin,
+        task::{Context, Poll},
+        time::{Duration, Instant},
+    },
+};
+
+/// Target specified in [`metrics::Metadata`] for all metrics produced by this
+/// crate.
+pub const METADATA_TARGET: &str = "future_metrics";
+
+/// Metric names used by this crate.
+pub mod metric_name {
+    pub const FUTURE_DURATION: &str = "future_duration";
+
+    pub const FUTURES_CREATED: &str = "futures_created";
+    pub const FUTURES_STARTED: &str = "futures_started";
+    pub const FUTURES_FINISHED: &str = "futures_finished";
+    pub const FUTURES_CANCELLED: &str = "futures_cancelled";
+
+    pub const FUTURE_POLL_DURATION: &str = "future_poll_duration";
```

Comment:
What's the difference between `poll_duration` and `poll_duration_total`?

Reply:
`poll_duration` - duration of a single poll
`poll_duration_total` - sum of all poll durations for a single future

Reply:
I see. With `opentelemetry` we would just calculate the average by dividing the total by the number of poll entries. Not sure though how these fancy histograms would work.